cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ExecutionTaskTracker.java

```diff
@@ -0,0 +1,244 @@
+      tasksInProgress.addAll(_inProgressTasks.get(balancingAction));
+    }
+
+    return tasksInProgress;
+  }
+
+  /**
+   * Get all the aborted execution tasks.
+   */
+  public Set<ExecutionTask> tasksAborted() {
+    Set<ExecutionTask> tasksAborted = new HashSet<>();
+    for (BalancingAction balancingAction : BalancingAction.values()) {
+      tasksAborted.addAll(_abortedTasks.get(balancingAction));
+    }
+
+    return tasksAborted;
+  }
+
+  /**
+   * Get all the dead execution tasks.
+   */
+  public Set<ExecutionTask> tasksDead() {
+    Set<ExecutionTask> tasksDead = new HashSet<>();
+    for (BalancingAction balancingAction : BalancingAction.values()) {
+      tasksDead.addAll(_deadTasks.get(balancingAction));
+    }
+
+    return tasksDead;
+  }
+
+  public int numDeadReplicaMove() {
+    return _deadTasks.get(BalancingAction.REPLICA_MOVEMENT).size();
+  }
+
+  public int numDeadLeadershipMove() {
+    return _deadTasks.get(BalancingAction.LEADERSHIP_MOVEMENT).size();
+  }
+
+  public int numDeadReplicaAddition() {
+    return _deadTasks.get(BalancingAction.REPLICA_ADDITION).size();
+  }
+
+  public int numDeadReplicaDeletion() {
+    return _deadTasks.get(BalancingAction.REPLICA_DELETION).size();
+  }
+
+  public int numAbortedReplicaMove() {
+    return _abortedTasks.get(BalancingAction.REPLICA_MOVEMENT).size();
+  }
+
+  public int numAbortedLeadershipMove() {
+    return _abortedTasks.get(BalancingAction.LEADERSHIP_MOVEMENT).size();
+  }
+
+  public int numAbortedReplicaAddition() {
+    return _abortedTasks.get(BalancingAction.REPLICA_ADDITION).size();
+  }
+
+  public int numAbortedReplicaDeletion() {
+    return _abortedTasks.get(BalancingAction.REPLICA_DELETION).size();
+  }
+
+  public int numInProgressReplicaMove() {
+    return _inProgressTasks.get(BalancingAction.REPLICA_MOVEMENT).size();
+  }
+
+  public int numInProgressLeadershipMove() {
+    return _inProgressTasks.get(BalancingAction.LEADERSHIP_MOVEMENT).size();
+  }
+
+  public int numInProgressReplicaAddition() {
+    return _inProgressTasks.get(BalancingAction.REPLICA_ADDITION).size();
+  }
+
+  public int numInProgressReplicaDeletion() {
+    return _inProgressTasks.get(BalancingAction.REPLICA_DELETION).size();
+  }
+
+  public int numPendingReplicaMove() {
+    return _pendingProposals.get(BalancingAction.REPLICA_MOVEMENT).size();
+  }
+
+  public int numPendingLeadershipMove() {
+    return _pendingProposals.get(BalancingAction.LEADERSHIP_MOVEMENT).size();
+  }
+
+  public int numPendingReplicaAddition() {
+    return _pendingProposals.get(BalancingAction.REPLICA_ADDITION).size();
+  }
+
+  public int numPendingReplicaDeletion() {
+    return _pendingProposals.get(BalancingAction.REPLICA_DELETION).size();
+  }
+
+  /**
+   * Clear (1) dead tasks, (2) aborted tasks, (3) in progress tasks, and (4) pending proposals.
+   */
+  public void clear() {
+    // Clear dead tasks.
+    for (BalancingAction balancingAction : BalancingAction.values()) {
```

Comment:
There is no need to clear each of the sets in the Map. _deadTasks.clear() should be sufficient.